src/services/recipient.js

```diff
@@ -889,20 +889,23 @@ export async function getGoalsByActivityRecipient(
     },
   });
 
+  // For checkbox selection we only need the primary goal id.
+  const rolledUpGoalIds = r.goalRows.map((goal) => goal.id);
+
   if (limitNum) {
     return {
       count: r.goalRows.length,
       goalRows: r.goalRows.slice(offSetNum, offSetNum + limitNum),
       statuses,
-      allGoalIds,
+      allGoalIds: rolledUpGoalIds,
```

Comment:
Doesn't this mean that the places where we use allGoalIds on the frontend will now be missing goals? Print goals, I think, for example

Reply:
Looking at the history I think I added this for this exact purpose. I tested a case when I select a goal that is rolled up (ie has two goals) then clicked print. Both goals show up in the print preview but rolled up.

Reply:
![image](https://github.com/HHS/Head-Start-TTADP/assets/84350609/c8506bef-32eb-4602-965a-c96eb7b2ac5e)